crates/primitives/src/snapshot/segment.rs

```diff
@@ -32,49 +50,47 @@ impl SnapshotSegment {
     }
 
     /// Returns the default file name for the provided segment and range.
-    pub fn filename(&self, range: &RangeInclusive<BlockNumber>) -> PathBuf {
-        let (filters, compression) = self.config();
-        self.filename_with_configuration(filters, compression, range)
+    pub fn filename(&self, range: &RangeInclusive<BlockNumber>) -> String {
+        // ATTENTION: if changing the name format, be sure to reflect those changes in
+        // [`Self::parse_filename`.]
+        format!("snapshot_{}_{}_{}", self.as_ref(), range.start(), range.end(),)
     }
 
-    /// Returns file name for the provided segment, filters, compression and range.
+    /// Returns file name for the provided segment and range, alongisde filters, compression.
     pub fn filename_with_configuration(
         &self,
         filters: Filters,
         compression: Compression,
         range: &RangeInclusive<BlockNumber>,
-    ) -> PathBuf {
-        let segment_name = match self {
-            SnapshotSegment::Headers => "headers",
-            SnapshotSegment::Transactions => "transactions",
-            SnapshotSegment::Receipts => "receipts",
-        };
+    ) -> String {
+        let prefix = self.filename(range);
+
         let filters_name = match filters {
             Filters::WithFilters(inclusion_filter, phf) => {
-                let inclusion_filter = match inclusion_filter {
-                    InclusionFilter::Cuckoo => "cuckoo",
-                };
-                let phf = match phf {
-                    PerfectHashingFunction::Fmph => "fmph",
-                    PerfectHashingFunction::GoFmph => "gofmph",
-                };
-                format!("{inclusion_filter}-{phf}")
+                format!("{}-{}", inclusion_filter.as_ref(), phf.as_ref())
             }
             Filters::WithoutFilters => "none".to_string(),
         };
-        let compression_name = match compression {
-            Compression::Lz4 => "lz4",
-            Compression::Zstd => "zstd",
-            Compression::ZstdWithDictionary => "zstd-dict",
-            Compression::Uncompressed => "uncompressed",
-        };
 
-        format!(
-            "snapshot_{segment_name}_{}_{}_{filters_name}_{compression_name}",
-            range.start(),
-            range.end(),
-        )
-        .into()
+        // ATTENTION: if changing the name format, be sure to reflect those changes in
```

Comment:
can we add a test to check that the result of `filename_with_configuration` can be parsed by `parse_filename`?